get_png_files_from_drive.py

```diff
@@ -0,0 +1,200 @@
+    merge_png_dirs(output_dir)
+
+
+def get_service():
+    """Autenticate yourself and create a token.pickle to use the google apiclient.
+    The file token.pickle stores the user's access and refresh tokens, and is
+    created automatically when the authorization flow completes for the first
+    time."""
+
+    creds = None
+    if path.isfile("token.pickle"):
+        with open("token.pickle", "rb") as token:
+            creds = pickle.load(token)
+    # If there are no (valid) credentials available, let the user log in.
+    if not creds or not creds.valid:
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            if not path.isfile("credentials.json"):
+                sys.exit("You are missing 'credentials.json'. "
+                         "Please get this file here: "
+                         "https://developers.google.com/drive/api/v3/quickstart/python "
+                         "and include it in the root of your project.")
+            flow = InstalledAppFlow.from_client_secrets_file("credentials.json", SCOPES)
+            creds = flow.run_local_server(port=0)
+        # Save the credentials for the next run
+        with open("token.pickle", "wb") as token:
+            pickle.dump(creds, token)
+
+    service = build("drive", "v3", credentials=creds)
+
+    return service
+
+
+def get_folder_id(service, folder_name):
+    """Get the folder id instead of the folder name. The api needs an
+    ID instead of a Name"""
+    folder = service.files().list(
+            q=f"name='{folder_name}' and mimeType='application/vnd.google-apps.folder'",
+            fields='files(id, name, parents)').execute()
+
+    total = len(folder['files'])
+
+    if total != 1:
+        sys.exit(f'{total} folders found, needs exactly one')
+
+    folder_id = folder['files'][0]['id']
+
+    return folder_id
+
+
+def ensure_directory_exists(dir_name):
+    if not path.exists(dir_name):
+        makedirs(dir_name)
+
+    return dir_name
+
+
+def get_file_list(service, folder_id):
+    """Get all files in the Google drive folder.
+    So they can be downloaded"""
+    result = []
+    page_token = None
+    while True:
+        files = service.files().list(
+                q=f"'{folder_id}' in parents",
+                fields='nextPageToken, files(id, name, mimeType)',
+                pageToken=page_token,
+                pageSize=1000).execute()
+
+        result.extend(files['files'])
+
+        page_token = files.get("nextPageToken")
+
+        if not page_token:
+            break
+    return result
+
+
+def download_files(service, file_list, output_dir):
+    """Download all the files in the file_list and
+    place them in the output_dir"""
+    print("Downloading files")
+    for file in file_list:
+        file_id = file['id']
+        filename = file['name']
+
+        if "emoji_u" in filename and ".png" in filename:
```

Comment:
drive typically has only AI files

Reply:
Since we can't convert the AI files to PNG without Illustrator, is a folder with PNG artwork a valid startingpoint. We can do the conversion before running this script.